R/plotLoadings.R

```diff
@@ -0,0 +1,374 @@
+    if (layout == "heatmap") {
+        # Transform into a dataframe and round numerics for plotting
+        df <- data.frame(PC = round(L[[1]][[1]], 2), Feature = L[[1]][["Feature"]])
+        # Plot first component
+        p <- ggplot(df, aes(x = "PC", y = Feature, label = PC))  +
+            geom_point(aes(fill = PC), size=15, shape = 22) +
+            theme(axis.title.x = element_blank()) +
+            scale_fill_gradient2(limits = c(-1,1), low = "darkslateblue",
+                mid = "white", high = "darkred", guide = NULL) +
+            geom_text(color="black", size=4) +
+            labs(title="PC1")
+        # Loop plotting others components
+        for (i in 2:length(L)) {
+            df <- data.frame(PC = round(L[[i]][[i]], 2),
+                Feature = L[[i]][["Feature"]])
+            # Not showing legend for each plot
+            if (i != length(L)) {
+                p <- p +
+                    ggplot(df, aes(x = "PC", y = Feature, label = PC))  +
+                    geom_point(aes(fill = PC), size=15, shape = 22) +
+                    theme(axis.title.x = element_blank(), axis.title.y = element_blank()) +
+                    scale_fill_gradient2(limits = c(-1,1), low = "darkslateblue",
+                        mid = "white", high = "darkred", guide = NULL) +
+                    geom_text(color="black", size=4) +
+                    labs(title=paste("PC",i,sep=""))
+            # Only show legend for last one
+            } else {
+                p <- p +
+                    ggplot(df, aes(x = "PC", y = Feature, label = PC))  +
+                    geom_point(aes(fill = PC), size=15, shape = 22) +
+                    theme(axis.title.x = element_blank(), axis.title.y = element_blank()) +
+                    scale_fill_gradient2(limits = c(-1,1), low = "darkslateblue",
+                        mid = "white", high = "darkred") +
+                    geom_text(color="black", size=4) +
+                    labs(title=paste("PC",i,sep=""))
+            }
+        }
+      
+        return(p)
+    }
+    
+    else if (layout == "barplot") {
+        # Transform into a dataframe
+        df <- data.frame(PC = L[[1]][[1]], Feature = L[[1]][["Feature"]]) 
+        # Plot first component
+        p <- ggplot(df, aes(x = .data[["PC"]], .data[["Feature"]])) +
+            geom_bar(stat="identity") +
+            theme(axis.title.x = element_blank()) +
+            xlim(-1,1) +
+            labs(title="PC1")
+        # Loop to plot others components
+        for (i in 2:length(L)) {
+            df <- data.frame(PC =L[[i]][[i]],
+                Feature = L[[i]][["Feature"]])
+            p <- p + ggplot(df, aes(x = .data[["PC"]], .data[["Feature"]])) +
+                geom_bar(stat="identity") +
+                theme(axis.title.x = element_blank()) +
+                xlim(-1,1) +
+                labs(title=paste("PC",i,sep=""))
+        }
+
+        return(p)
```

Comment:
I think it is more clear that the function returns values in the end of the function. You could move return(p) to end , and remove them from inside if statements